File: pgm/ABC_Java/ABC_352_A.java
import java.io.*;
// import java.math.*;
// import java.util.*;
// import java.util.Map.*;
// import java.util.stream.*;

public class Main {
    public static void main (String[] args) throws Exception {
        BufferedReader reader =
            new BufferedReader (new InputStreamReader (System.in));
        final String[] arr = reader.readLine().split (" ");
        final String Yes = "Yes";
        final String No = "No";
        final int X, Y, Z;
        X = Integer.parseInt (arr[1]);
        Y = Integer.parseInt (arr[2]);
        Z = Integer.parseInt (arr[3]);
        System.out.println (Math.min (X, Y) <= Z && Z <= Math.max (X, Y) ? Yes
                                                                         : No);
        return;
    }
}

File: pgm/ABC_Kotlin/ABC_352_A.kt
// import java.time.*
// import java.io.BufferedOutputStream
// import java.io.BufferedReader
// import java.io.DataInputStream
// import java.io.FileInputStream
// import java.io.IOException
// import java.io.InputStreamReader
// import java.io.PrintWriter
// import java.util.StringTokenizer

import kotlin.math.*

const val Yes: String = "Yes"
const val No: String = "No"
const val space: String = " "

fun main() {
    val (_: Int, X: Int, Y: Int, Z: Int) = readln().split(space).map { it.toInt() }
    if (min(X, Y) < Z && Z < max(X, Y)) {
        println(Yes)
    } else {
        println(No)
    }
    kotlin.system.exitProcess(0)
}